Update the Finnish translation file to fix some typos and add missing…

ZeiP opened the pull request:
… translations.

fellen commented:
Thanks! I have asked the last translator to comment. Remind me, if we get no answer in acceptable time.

kohtala commented on po/fi.po:
```diff
@@ -6305,26 +6252,26 @@ msgid ""
 msgstr ""
-"Tätä tiliä ei voi muokata. Jos haluat muokata tilikirjan tositteita, avaa "
-"tilin asetukset ja poista koontitilin valinta."
+"Tämän tilin tapahtumia ei voi muokata.\n"
```
tositteita

kohtala reviewed (changes requested):
I glanced over. Mostly good and big thanks for great work.

Some deviations from glossary. Translations of split to jako should be changed to rivi. Invoiced should make clear which kind of invoice it is. Transaction to tapahtuma should be changed to tosite.

These are what I noticed with the short time I had for this.

ZeiP commented on po/fi.po:
```diff
@@ -3476,23 +3467,20 @@ msgid ""
-#, fuzzy
 msgid "Copy an existing Budget."
-msgstr "Kopioi olemassa oleva budjetti"
+msgstr "Kopioi olemassaoleva budjetti."
```
The original was marked as fuzzy, which is why I changed it, but it indeed seems you're right.

ZeiP commented on po/fi.po:
```diff
@@ -1543,11 +1543,11 @@ msgstr ""
 #: gnucash/gnome/gnc-plugin-page-register.c:631
 msgid "Manage Document Link"
-msgstr ""
+msgstr "Muuta liittyvää asiakirjaa"
```
Yeah, it's used at least in the register. You can add/change the document that's attached to the transaction. IMO the translation already works for that, but if you have a better one in mind feel free to suggest another phrasing.

kohtala replied:
"Muuta linkkiä asiakirjaan"? "Muuta asiakirjalinkkiä" maybe too monstrous word?

ZeiP replied:
IMO it's a bit technical to refer to the link – for a normal user a link means mainly a link to an URL. Instead I'd suggest ”Vaihda yhdistettyä asiakirjaa”, is that ok? I also thought I'd change all the ”liittyvä asiakirja” to ”yhdistetty asiakirja” as that's a bit closer to the source text and also seems a bit more fluent.

fellen replied:
You can manage the links to your documents, not the documents – we don't know which app is associated on your computer.
I have attached a screenshot to glossary/document link.

ZeiP commented on po/fi.po:
```diff
@@ -2561,40 +2557,40 @@ msgstr ""
 #: gnucash/gnome/dialog-order.c:878
 msgid "Owner Name "
-msgstr "Omistajan nimi"
+msgstr "Omistajan nimi "
```
That's because there's also a space in the msgid.

fellen replied:
> That's because there's also a space in the msgid.

In such a case add a comment on the source  string. Done by me now.

ZeiP commented on po/fi.po:
```diff
@@ -5582,16 +5541,12 @@ msgid "_Sort By..."
-#| msgid "Do not print transactions out of specified dates."
 msgid "Move to the split at the specified date"
-msgstr "Älä näytä tositteita määritettyjen päivien ulkopuolelta."
+msgstr "Siirrä jako annetulle päivälle"
```
The glossary isn't visible in Poedit, and jako is used in quite many places around the translation, so it seems an honest mistake to make. Perhaps the next step after this translation is accepted is to go through these common mistakes and fix them all.

ZeiP commented on po/glossary/fi.po:
```diff
@@ -579,19 +576,19 @@ msgstr "maksu"
 #. "An account where no transactions may be posted to; transactions can only be posted to subaccounts of this account, so this account serves as a placeholder in the hierarchy"
 msgid "placeholder"
-msgstr "koontitili"
+msgstr "paikanpitäjä"
```
Why? AFAIK it's not an account that accumulates all the transactions in the accounts below, but rather one that just makes up the hierarchy as per the comment. I admit paikanpitäjä isn't ideal either, though. Maybe ”luokkatili” or similar?

kohtala replied:
It collects the accounts even when sums are not there.
I have tried to find time to move my accounting to gnucash, but have not yet found the time. Since I've not used it much, I could get something wrong so it is good you bring up your opinion. I believe some reports go the account tree down only to given depth an do show the sums also for all the accouns under placeholder.
@harski translated this earlier. Maybe we can get a third opinion.

ZeiP replied:
Well, kinda, yeah. Reports do of course in some cases show up sums of the subtrees, but I wouldn't call it a feature of the parent account. Also the English term + the comment above it IMO would match much more closely to my suggestion (which, I'll admit, is not too pretty).

I actually wonder if this is something that should also be changed in the English version – IMO a better English word would also be hierarchy account or similar, I remember when I was beginning my use of GnuCash some ten years ago and I think I had some trouble with understanding what this term meant.

If this is the only blocker, then maybe I'll just revert it and we'll get back to it. This isn't even an UI translation but one for the glossary, so it won't change anything anyway.

fellen replied:
>  I believe some reports go the account tree down only to given depth an do show the sums also for all the accouns under placeholder.

No, there is no limit in the data structure, but some reports can filter details by level.

fellen replied:
> I actually wonder if this is something that should also be changed in the English version – IMO a better English word would also be hierarchy account or similar, I remember when I was beginning my use of GnuCash some ten years ago and I think I had some trouble with understanding what this term meant.

As you are using weblate, you can file suggestions and comments there against the english text. As we are pretty new there, We still have to work on the process, to improve the source text.

ZeiP replied:
I haven't been using Weblate so far, I created the translation using Poedit. But thanks for the tip, I'll maybe suggest this there.

harski replied:
This is a tough one. I agree with @ZeiP that the original English term `placeholder` here might not be that accurate in the first place and should be considered to be changed. But for this change set it is what it is, and the discussion for the English term should be had elsewhere.

While I also agree with the principle that the translation shouldn't diverge much from the original text, I don't think the proposed translation `paikanpitäjä`, is a good one. It doesn't sound natural in Finnish *and* it doesn't describe the option well.

> AFAIK it's not an account that accumulates all the transactions in the accounts below, but rather one that just makes up the hierarchy as per the comment.

I understand `koontitili` like @kohtala, something that groups the sub-accounts together. I'm not against changing it, though, if you come up with a better term @ZeiP. What is your reasoning for `luokkatili`?

If we don't come up with a better translation, I vote for leaving this as is.

fellen replied:
> I agree with @ZeiP that the original English term `placeholder` here might not be that accurate in the first place and should be considered to be changed.

I am inclined to change this attribute to "write protected" or "read-only" after the string freeze is lifted again. That is the real meaning. The original coder tried to be user friendly and saw the use case "group similar accounts in one category". Later users discovered it can be used to write-protect closed accounts, too.

kohtala replied:
Thanks for the insight.

ZeiP commented:
> I glanced over. Mostly good and big thanks for great work.
> 
> Some deviations from glossary. Translations of split to jako should be changed to rivi. Invoiced should make clear which kind of invoice it is. Transaction to tapahtuma should be changed to tosite.
> 
> These are what I noticed with the short time I had for this.

Yeah. The glossary defines some translations that aren't really populated through the actual translation, which makes it less efficient. I fixed all the other existing uses of splits and transactions also, maybe this'll make it easier in the future to stick to the correct terms. The glossary is difficult to use as at least in my Poedit I couldn't make it show up at all (and I noticed it fairly late to my translation process).

I reverted a few of the changes I wasn't sure of where you asked for more checks, as I currently don't have the time to check on those – I intend this to be a first bump towards a fuzzy-free and otherwise more comprehensive translation, but I might be adding some new translations in the future if I have time, there's still lots to do. I also removed those fuzzies that didn't match the English version at all but which I didn't have the time or knowledge to translate properly.

Otherwise I made almost all the changes you asked for, there are a few more comments which I answered but didn't fix yet.

kohtala reviewed (commented):
This is better. Definitely a lot of improvement. The placeholder translation is something I'd like to check. If @harski does not respond soon to approve your suggestion, could you revert that and we can get back to it later?
I think you noticed I had commented on the glossary likeit was used. I realized I had done that soon after I left. Just shows I was busy trying get you some feedback to get your contribution going forward. Thanks a lot.

ZeiP commented:
No problem, and thanks for the quick feedback! I reverted the placeholder for now per the discussion in the thread. I also changed the linked documents as I described in the other thread. Is there still something that should be improved upon for now?

fellen commented:
> If @harski does not respond soon to approve your suggestion, could you revert that and we can get back to it later?

I can give both of you review rights for GnuCash at weblate, if you give me your weblate names. I imagine one edits and the other approves.

ZeiP commented:
> > If @harski does not respond soon to approve your suggestion, could you revert that and we can get back to it later?
> 
> I can give both of you review rights for GnuCash at weblate, if you give me your weblate names. I imagine one edits and the other approves.

I just created an account with the username ZeiP (email jyri-petteri.paloposki@iki.fi.)

fellen commented:
> I just created an account with the username ZeiP (email [jyri-petteri.paloposki@iki.fi](mailto:jyri-petteri.paloposki@iki.fi).)

Added as reviewer.
But to avoid collisions, you should start editing at weblate – at least strings changed in this PR – after I have pushed this PR.
Please confirm, this PR is finished.

ZeiP commented:
> Added as reviewer.
> But to avoid collisions, you should start editing at weblate – at least strings changed in this PR – after I have pushed this PR.
> Please confirm, this PR is finished.

Great, thanks! Yeah, I assumed so. On my side the PR is ready now.

fellen commented:
```
msgfmt -c --statistics --check-accelerators="_" po/fi.po
po/fi.po:15805: msgstr lacks the keyboard accelerator mark '_'
msgfmt: found 1 fatal error
3557 translated messages, 1988 untranslated messages.
```
But you can fix it later.
Thanks!

kohtala commented:
Thanks all. Great to have the contribution.

I created account @kohtala at weblate.

fellen commented:
> I created account @kohtala at weblate.

Added
